KeyedDecodingContainer.swift

```diff
@@ -8,291 +8,214 @@
 
 import Foundation
 
+extension String {
+    
+    // MARK: - Date Formtter
+    
+    func date(format: String) -> Date? {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .none
+        dateFormatter.timeStyle = .none
+        dateFormatter.timeZone = .autoupdatingCurrent
+        dateFormatter.dateFormat = format
+        return dateFormatter.date(from: self)
+    }
+}
+
 extension KeyedDecodingContainer {
-    //MARK: - String
+    
+    // MARK: - String
+    
     func decodeIfPresent(_ type: String.Type, forKey key: K) throws -> String? {
-        var value:String?
-        do {
-            value = try self.decode(String.self, forKey: key)
-        }catch{
-            do {
-                let boolValue = try self.decode(Bool.self, forKey: key)
-                value = String(boolValue)
-            }catch{
-                do {
-                    let intValue = try self.decode(Int.self, forKey: key)
-                    let floatValue = try self.decode(Float.self, forKey: key)
-                    if (Float(intValue) == floatValue) {
-                        value = String(intValue)
-                    }else {
-                        value = String(floatValue)
-                    }
-                }catch{
-                }
-            }
+        if let stringValue = try? decode(String.self, forKey: key) {
+            return stringValue
+        } else if let boolValue = try? decode(Bool.self, forKey: key) {
+            return String(boolValue)
+        } else if let intValue = try? decode(Int.self, forKey: key) {
+            return String(intValue)
+        } else if let doubleValue = try? decode(Double.self, forKey: key) {
+            return String(doubleValue)
+        } else {
+            return "-"
```

Comment:
@AnasAlhasani  I prefer to return nil if we can't handle the value, this is the main idea for optional value to let developer know the value did not parsed correctly here, if we return default value we will change the result and that will make so much issues for developer